### PR TITLE
feat(derive_code_mappings): Include repository files for various languages

### PR DESCRIPTION
**Pull request by armenzg**
This change does not add support for multiple languages for code derivation, however, it start keeping track of the source files for these languages:

* Javascript (.js)
* React (.jsx)
* Typescript + React (.tsx)
* Typescript (.ts)
* Node ES6 modules (.mjs)
* Python (.py)
* Ruby (.rb)
* Php (.php)
* Go lang (.go)

Follow up changes will be required to add code mapping derivation for these languages.

Fixes WOR-2427

### CODE REVIEW

File: src/sentry/integrations/github/client.py [hunk 1]
```diff
@@ -70,10 +70,18 @@ def get_repo(self, repo: str) -> JSONData:
         repository: JSONData = self.get(f"/repos/{repo}")
         return repository
 
-    # https://docs.github.com/en/rest/git/trees#get-a-tree
-    def get_tree(self, repo_full_name: str, tree_sha: str) -> List[str]:
-        tree = []
+    def get_repo_files(
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:73]: I'm changing this function a bit to only focus on return files for a repo. There's an option to return all files or just source code files.

File: src/sentry/integrations/github/client.py [hunk 2]
```diff
@@ -71,9 +71,10 @@ def get_repo(self, repo: str) -> JSONData:
         return repository
 
     # https://docs.github.com/en/rest/git/trees#get-a-tree
-    def get_tree(self, repo_full_name: str, tree_sha: str) -> List[str]:
-        tree = []
+    def get_tree(self, repo_full_name: str, tree_sha: str) -> JSONData:
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:74]: Moving the logic of extracting files from a tree to `get_repo_files`, thus, return type change.

File: src/sentry/integrations/github/client.py [hunk 3]
```diff
@@ -88,7 +89,7 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> List[str]:
                 logger.warning(
                     f"The tree for {repo_full_name} has been truncated. Use different a approach for retrieving contents of tree."
                 )
-            tree = trim_tree(contents["tree"], ["python"])
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:91]: We move the responsibility of flattening the tree to a list of files to `get_repo_files`.

File: src/sentry/integrations/github/client.py [hunk 4]
```diff
@@ -103,6 +104,27 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> List[str]:
...
+
+        repo_full_name: e.g. getsentry/sentry
+        tree_sha: A branch or a commit sha
+        only_source_code_files: Include all files or just the source code files
+        """
+        repo_files = []
+        try:
+            tree = self.get_tree(repo_full_name, tree_sha)
+            if tree is not None:
+                repo_files = [x["path"] for x in tree if x["type"] == "blob"]
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:120]: This is the line that flattens the tree.

File: src/sentry/integrations/github/client.py [hunk 5]
```diff
@@ -103,6 +104,27 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> List[str]:
...
+                if only_source_code_files:
```
Comment (armenzg) [inline src/sentry/integrations/github/client.py:121]: For now, we only use this function with the default value but we may in the future call it with a False value.

**Review by NisanthanNanthakumar — APPROVED** [review]

**Pull request merged**